test/jit/test_with.py

```diff
@@ -610,3 +609,28 @@ def forward(self, x, y):
         w = s(x, y)
 
         self.assertFalse(w.requires_grad)
+
+    def test_with_record_function(self):
+        """
+        Check that torch.autograd.profiler.record_function context manager is
+        torchscriptable.
+        """
+        def with_rf(x, y):
+            # type: (Tensor, Tensor) -> Tensor
+            with torch.autograd.profiler.record_function("foo"):
```

Comment:
Consider testing nested `record_function` calls to make sure the context managers are nested properly. There are tests for this already with a much simpler context manager though so it's not a must-have.

Reply:
Added this test.